vault/resource_token.go

```diff
@@ -89,6 +89,12 @@ func tokenResource() *schema.Resource {
 				Default:     "token",
 				Description: "The display name of the token.",
 			},
+			"metadata": {
```

Comment:
I like that you made this more readable by extending its name. I also think it could easily confuse people looking for the `"meta"` field specifically. Could we rename this to `"meta"` to exactly match the field this maps to in the Vault API?